### PR TITLE
Improve Calendar form field style - Issues #36933 &  #33447

### PR DESCRIPTION
**Pull request by cyrez**
Pull Request for Issues #36933 &  #33447.

### Summary of Changes
Improve display of datetime picker.
- Centering calendar table for better looking in some not-english language.
- Style time select fields.
- Fix footer buttons alignment.
- Improve display of time fields depending on field settings (weeknumbers, am/pm).

### Testing Instructions
- Apply patch, and test any form field of type calendar. 

### Actual result BEFORE applying this Pull Request
before-patch-fr
_Example in French_
 
 
### Expected result AFTER applying this Pull Request
after-patch-fr
_Example in French_
 
 
### Rendering depending on calendar field settings:
With or without weeknumbers, am/pm and or time. And with RTL support

_Legend:_
1. Week numbers, time 24h
2. Week numbers, time 12h
3. Time 24h
4. Time 12h
5. No Time
6. RTL (arabic)
 

(1) weeknumber-time-24h (2) weeknumber-time-12h
(3) time-24h (4) time-12h
(5) weeknumbers-notime (6) rtl-support

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: build/media_source/system/css/fields/calendar-rtl.css [hunk 1]
```diff
@@ -1,10 +1,12 @@
 /**
- * @copyright  (C) 2016 Open Source Matters, Inc. <https://www.joomla.org>
+ * @copyright  (C) 2016-2023 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline build/media_source/system/css/fields/calendar-rtl.css]: Please revert this. As of about 18 months agon the decision was made to only use the first date of copyright (as you will see in every other file)

Comment (cyrez) [reply build/media_source/system/css/fields/calendar-rtl.css]: Ok! 👍

File: build/media_source/system/css/fields/calendar.css [hunk 2]
```diff
@@ -1,10 +1,12 @@
 /**
- * @copyright  (C) 2016 Open Source Matters, Inc. <https://www.joomla.org>
+ * @copyright  (C) 2016-2023 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline build/media_source/system/css/fields/calendar.css]: please revert

**Comment by cyrezdev** [comment]
> I am confused. If I understand correctly then the problem only exists in non-english languages where the text for "today" is long. If that is correct then what is the purpose of all the screenshots you have posted in english

As mentionned in the description of this PR, it's not only for this point, but 3 other ones:
- To style time select fields.
- Fix footer buttons alignment.
- Improve display of time fields depending on field settings (weeknumbers, am/pm).

So the additionnal screenshots are valid to show the different possible settings for calendar form field.

It is mainly cosmetic here ;-)

File: build/media_source/system/css/fields/calendar-rtl.css (hunk 1 above)
Comment (cyrez) [reply build/media_source/system/css/fields/calendar-rtl.css]: Done.

Thanks!

**Comment by Chaosxmk** [comment]
Looks good, fixes the main point of issue I highlighted.

**Comment by obuisard** [comment]
> Looks good, fixes the main point of issue I highlighted.

@Chaosxmk, please help test the PR, we need two successful tests to get this great improvement merged into the project :-)

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on b668fc252b101fbf1115e96bcdd8b1a37cb6898c<br /><br />Tested with English and Persian<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40761">issues.joomla.org/tracker/joomla-cms/40761</a>.</sub>

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 145606ed79c9587819933b13ee2f7f1192a6e344<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40761">issues.joomla.org/tracker/joomla-cms/40761</a>.</sub>

**Comment by obuisard** [comment]
Great job Cyril @cyrezdev, much needed improvements and now we have a decent calendar field :-)

**Comment by cyrezdev** [comment]
> Great job Cyril @cyrezdev, much needed improvements and now we have a decent calendar field :-)

Thank you Olivier!

**Comment by cyrezdev** [comment]
Hello here!

This PR has introduced an issue with no possibility to set Time with calendar picker, if the week numbers are hidden and the time format is 24h.

Here a quick PR with the patch: https://github.com/joomla/joomla-cms/pull/42183

Thank you for testing! 👍

**Pull request merged**